crates/extensions/c8y_http_proxy/src/messages.rs

```diff
@@ -40,6 +40,12 @@ pub enum C8YRestError {
 
     #[error(transparent)]
     FromIOError(#[from] std::io::Error),
+
+    // `Display` impl of `C8yRestError` is used as part of an error message sent to the cloud in a smartrest message.
+    // Using `{anyhow::Error:?}` also prints the lower-level cause, so using it here will result in a more detailed
+    // error message being sent to the cloud
+    #[error("Unexpected error: {0:?}")]
+    Unexpected(#[from] anyhow::Error),
```

Comment:
I don't love the name `Unexpected`. This implies the other possible error values in `C8YRestError` are "expected", which I think isn't remotely true for a lot of them. `FromAnyhow` seems like a better fit here.

I know this isn't related to the current changes you've made, but I'm not a fan of using the `Error` suffix in the variant names. This is partially due to the fact the enum name will (almost) always be present when referring to the variant name, and often it's possible to come up with a name that actually implies something has gone wrong (and thus describes itself better). The example I always think of is Java's `NullPointerException`, which IMO should be called `NullPointerDereference`, as that (a) explains what actually went wrong, and (b) like in Rust, it's completely obvious from context that you're dealing with an exception (in Java it's the presence of `throw` or `catch`, in Rust it's probably in a `Result` and probably in an enum called `SomethingError`).

Reply:
I agree with James that the name `Unexpected` is inappropriate and telling the user that the error was unexpected is useless.

About the `Error` suffix on error cases, this also makes sense ... an error should name the failed action. To keep in mind for future error cases!

Reply:
> I agree with James that the name `Unexpected` is inappropriate and telling the user that the error was unexpected is useless.

True, in this case almost all errors variants are unexpected, but some errors derived from the input, like validation failures, can be _expected_, and in this case _unexpected_ would be something that has no relation to the input or the data provided by the user. _Expected_ errors are never bugs, but unexpected errors can be bugs.

Reply:
> so would simply `Other` be okay?

I'm okay for that.